src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -247,6 +249,67 @@ void ManageSurfaceHeatBalance(EnergyPlusData &state)
 // Beginning Initialization Section of the Module
 //******************************************************************************
 
+Real64 truncateOverrideValue(Real64 value)
+{
+    if (value > 0) {
+        return max(min(value, 0.9999), 0.0001);
+    } else { // special value for illegal index
+        return value;
+    }
+}
+
+std::pair<Real64, Real64> GetSurfAddOnOverrideValue(EnergyPlusData &state,
+                                                    int surfNum,
+                                                    Material::VariableAbsCtrlSignal controlSignal,
+                                                    int scheduleIdxThermal,
+                                                    int functionIdxThermal,
+                                                    int scheduleIdxSolar,
+                                                    int functionIdxSolar)
+{
+    // 0.0 means invalid, will not be used in absorptance value override
+    Real64 overrideValueThermal = 0.0;
+    Real64 overrideValueSolar = 0.0;
+    if (controlSignal == Material::VariableAbsCtrlSignal::Scheduled) {
+        overrideValueThermal = ScheduleManager::GetCurrentScheduleValue(state, scheduleIdxThermal);
```

Comment:
One of these could be `-1`, no? Don't you have to check if they are `-1` the same way you are checking the function indices below?

Reply:
The scheduleIdx will be 0 if GetScheduleIndex didn't find it.
Then the GetCurrentScheduleValue will return 0 if the input index of the schedule is 0. 
So maybe the -1 case won't appear?

Reply:
Oh I initialized them in the struct as -1

Reply:
Right. I meant one of them could be 0. Shouldn’t you check for that?

Reply:
If the schedule index is 0, then the returned override value will also be 0. The following chunk will make sure to overwrite only when the value is positive
```
        std::pair<Real64, Real64> overrideValues = GetSurfAddOnOverrideValue(state,
                                                                             surfNum,
                                                                             thisMaterial->absorpVarCtrlSignal,
                                                                             thisMaterial->absorpThermalVarSchedIdx,
                                                                             thisMaterial->absorpThermalVarFuncIdx,
                                                                             thisMaterial->absorpSolarVarSchedIdx,
                                                                             thisMaterial->absorpSolarVarFuncIdx);
        if (overrideValues.first > 0.0) state.dataHeatBalSurf->SurfAbsThermalExt(surfNum) = overrideValues.first;
        if (overrideValues.second > 0.0) state.dataHeatBalSurf->SurfAbsSolarExt(surfNum) = overrideValues.second;
```

Reply:
Ok. I understand that you didn't create this behavior where if a schedule doesn't exist then you can still get a value out of it but that value is just zero, but this behavior is terrible, opaque, and leads to errors. Look at the gymnastics you have to do to deal with it, you have to clamp the value to 0.00001 and 0.99999 and then check that it's not 0.0 before you set it, and that logic is distributed over three different places. It's easier and clearer to handle this condition explicitly.

```cpp
if (thisMaterial->schedIdxSolar > 0) 
   state.dataHeatBalSurf->SurfAbsSolarExt(SurfNum) = std::clamp(GetCurrentScheduleValue(state, thisMaterial->schedIdxSolar), 0.0, 1.0);
```

Reply:
Oh! this way is nicer